extensions/wikia/EmbeddableDiscussions/scripts/DiscussionsSettingsModal.js

```diff
@@ -0,0 +1,223 @@
+
+			if (!$('#embeddable-discussions-settingsmodal-filter-by-all-checkbox').is(':checked')) {
+				// All not checked, populate with selected items in list
+				$selectedCategories.each(function() {
+					config.selectedCategories.push({ id: $(this).val() });
+				});
+			}
+
+			nirvana.sendRequest({
+				controller: 'EmbeddableDiscussionsApi',
+				method: 'saveCommunityPageSettings',
+				type: 'post',
+				format: 'json',
+				data: config,
+			});
+		}
+
+		function isSelected(id, selectedCategories) {
+			for (var i = 0; i < selectedCategories.length; i++) {
+				if (selectedCategories[i].id === id) {
+					return true;
+				}
+			}
+
+			return false;
+		}
+
+		function populateCategories(categories, selectedCategories) {
+			var $selectElement = $('#embeddable-discussions-settingsmodal-categories'), i;
+
+			if (selectedCategories.length === 0) {
+				$('#embeddable-discussions-settingsmodal-filter-by-all-checkbox').prop('checked', true);
+			}
+
+			$selectElement.change(function() {
+				// Reset checked status for All checkbox when any item in the category list is selected
+				$('#embeddable-discussions-settingsmodal-filter-by-all-checkbox').prop('checked', false);
+			});
+
+			for (i = 0; i < categories.length; i++) {
+				$selectElement.append($('<option>', {
+					value: categories[i].id,
+					text: categories[i].name,
+					selected: isSelected(categories[i].id, selectedCategories),
+				}));
+			}
+		}
+
+		function setSort(latest) {
+			if (latest) {
+				config.sortBy = 'latest';
+
+				$('#embeddable-discussions-settingsmodal-sort-latest').addClass(
+					'embeddable-discussions-settingsmodal-sort-selected'
+				);
+
+				$('#embeddable-discussions-settingsmodal-sort-trending').removeClass(
+					'embeddable-discussions-settingsmodal-sort-selected'
+				);
+
+			} else {
+				config.sortBy = 'trending';
+
+				$('#embeddable-discussions-settingsmodal-sort-trending').addClass(
+					'embeddable-discussions-settingsmodal-sort-selected'
+				);
+
+				$('#embeddable-discussions-settingsmodal-sort-latest').removeClass(
+					'embeddable-discussions-settingsmodal-sort-selected'
+				);
+			}
+		}
+
+		function openModal() {
+			$.when(
+				loader({
+					type: loader.MULTI,
+					resources: {
+						mustache: 'extensions/wikia/EmbeddableDiscussions/templates/SettingsModal.mustache',
+					}
+				}),
+				nirvana.sendRequest({
+					controller: 'EmbeddableDiscussionsApi',
+					method: 'getCommunityPageSettings',
+					type: 'get',
+					format: 'json'
+				})
+			).then(handleRequestsForModal);
+		}
+
+		/**
+		 * Handle messages, render modal and call createComponent
+		 * One of sub-tasks for getting modal shown
+		 * @param {Object} loaderRes
+		 * @param {Object} nirvanaRes
+		 */
+		function handleRequestsForModal(loaderRes, nirvanaRes) {
+			var categories;
+
+			config = nirvanaRes[0];
+
+			categories = config.allCategories;
+
+			// Pre-check all selected categories
+			for (i = 0; i < categories.length; i++) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the "let", "const" or "var" keyword to this declaration of "i" to make it explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AS2703)